src/transforms/mapperTransforms.ts

```diff
@@ -44,7 +46,9 @@ const pipe = (
   ...fns: Array<(pipelineValue: PipelineValue) => PipelineValue>
 ) => (x: PipelineValue) => fns.reduce((v, f) => (!v.isHandled ? f(v) : v), x);
 
-type ModelProperties = { [propertyName: string]: Mapper };
+let uberParents: string[] = [];
```

Comment:
I think this is a temporary solution, I don't love having this global array here. I think maybe we can have a set of "indexes" as part of ClientDetails that we can pass around as needed, this uberParents being one. I think we can index/store other stuff we are already processing to avoid double processing and increase performance. @daviwil  what do you think?

These improvements will probably come later once we have more functionality in or perf becomes an issue

Reply:
Is this something that could be generated and stored at the time we build `ClientDetails` if we also create a list of `MapperDetails` at the same time?  What I wonder is if we could build all the metadata we need in one step before the generation phase so that the generator just has to spit out code based on what is in the `*Details` objects.

At any rate, it'd be nice to have an issue to track the need for a better solution here, whatever it ends up being

Reply:
Makes sense I filed #538 to improve this